### PR TITLE
Added .pre-commit-hooks.yaml, fixes #8 and #9

### LINKED ISSUE #8: Do not remove last empty line (conflict with `end-of-file-fixer`!)
There is a conflict with the `end-of-file-fixer` pre-commit hook (https://pre-commit.com/hooks.html), which 

> ensures that a file is either empty, or ends with one newline.

But `nbstripout-fast` keeps removing the last empty line. (`nbstripout` doesn't)

### LINKED ISSUE #9: Do not overwrite file if contents are identical.
Otherwise, file metadata (last modified) in unnecessarily changed each time the script is used.

### PR DESCRIPTION
**Pull request by randolf-scholz**
- fixes #8 
- fixes #9 
- added `.pre-commit-hooks.yaml`
- fixed some typos and cleaned up [`examples/comparison.py`](https://github.com/deshaw/nbstripout-fast/blob/main/examples/comparison.py)

### CODE REVIEW

**Comment by randolf-scholz** [comment]
@mlucool Can this be merged?

File: src/main.rs [hunk 1]
```diff
@@ -155,18 +156,28 @@ fn process_file(
...
+    if cleaned_contents != *contents {
+        if let Some(file) = output_file {
+            fs::write(&file, cleaned_contents)
+                .map_err(|e| format!("Could not write to {:?} due to {:?}", file, e))?;
+        } else {
+            println!("{}", cleaned_contents);
+        }
     } else {
-        println!("{}", cleaned_contents);
+        println!("Content unchanged. File not modified.");
```
Comment (mlucool) [inline src/main.rs]: I'm not sure users want this written out in all cases of things not changed (especially if a file is passed in). Can we make this a `log::debug!` instead?

**Review by mlucool — CHANGES_REQUESTED** [review]
Thanks for the PR. Small change requested

**Comment by mlucool** [comment]
Thanks again for the PR!

Please [sign the CLA](https://github.com/deshaw/nbstripout-fast/blob/main/CONTRIBUTING.md) before we can merge your code.

**Review by mlucool — APPROVED** [review]

**Review by mlucool — APPROVED** [review]
Thanks again!

**Comment by mlucool** [comment]
Released as 1.0.2 on pypi

**Pull request merged**